### ISSUE TITLE
socketlabs connection errors lead to lost email

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-server#6730]
https://sentry.prod.mozaws.net/operations/olympia-prod/issues/672687/

So apparently `send_activity_mail` doesn't send email asynchronously which leads to lost email since those emails can't get retried. 

Last night we had connection issues to socketlabs and at least a few mails are lost because of that.

cc @eviljeff who implemented that function: I don't remember correctly why `send_activity_mail` doesn't pass the `async` flag to `send_mail` but maybe there was a reason? If not we should fix that.

```
error: [Errno 110] Connection timed out
(22 additional frame(s) were not displayed)
...
  File "olympia/editors/utils.py", line 510, in notify_email
    settings.EDITORS_EMAIL, unique_id, perm_setting=perm_setting)
  File "olympia/activity/utils.py", line 306, in send_activity_mail
    perm_setting=perm_setting, reply_to=[reply_to])
  File "olympia/amo/utils.py", line 286, in send_mail
    attachments=attachments)
  File "olympia/amo/utils.py", line 246, in send
    return send_email(*args, **kwargs)
  File "olympia/amo/tasks.py", line 32, in send_email
    result.send(fail_silently=False)

error: [Errno 110] Connection timed out
```

**Comment by EnTeQuAk** [comment]
https://sentry.prod.mozaws.net/operations/olympia-prod/issues/672672/

**Comment by EnTeQuAk** [comment]
https://sentry.prod.mozaws.net/operations/olympia-prod/issues/672653/

**Comment by EnTeQuAk** [comment]
https://sentry.prod.mozaws.net/operations/olympia-prod/issues/672649/

**Comment by eviljeff** [comment]
I guess it never occurred to me to override the default and set `async=True`.  We don't do it _anywhere_, that I can find, and I don't see that we did it before the switch to activity mail either.  I agree that we should though - perhaps just change the default in `amo.tasks.send_email` ?

**Comment by EnTeQuAk** [comment]
Yeah, I do agree that changing the default in `amo.tasks.send_email` is probably the best way forward. cc @diox in case he knows some cases where we don't want that but I can't think of any.

**Comment by diox** [comment]
I wasn't aware in wasn't async by default either. I guess the only good argument that can be made to keep it sync by default is that if it happens inside the request-response cycle, and is sync, and it fails, it aborts the entire transaction, which is good if the mail was super critical. But yeah, we're probably better off making it async.

**Comment by EnTeQuAk** [comment]
Jupp, I'd rather fix that specific problem with `on_commit` hooks once django 1.11 lands :)

**Comment by eviljeff** [comment]
@EnTeQuAk the `on_commit` hook problem was fixed a while back right?  (And the issue is otherwise good to fix?)

**Comment by eviljeff** [comment]
I guess we should still do this?

**Comment by diox** [comment]
Duplicate of https://github.com/mozilla/addons-server/issues/6868
